### PR TITLE
atomname methods can handle empty groups

### LINKED ISSUE #2879: Atomnames methods should handle empty group
## Expected behavior ##

`u.residues[[]]._get_prev_residues_by_resid()` returns an empty residue group. Same for

* _get_next_residues_by_resid
* psi_selections
* omega_selections
* chi1_selections

## Actual behavior ##

IndexError due to not checking length first.

```python
>>> u.residues[[]].omega_selections()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 862, in omega_selections
    nxtres = residues._get_next_residues_by_resid()
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 670, in _get_next_residues_by_resid
    u = residues[0].universe
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/groups.py", line 527, in __getitem__
    return self.level.singular(self.ix[item], self.universe)
IndexError: index 0 is out of bounds for axis 0 with size 0
```

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD,  GRO, PDB, TPR, XTC, TRR,  PRMncdf, NCDF

u = mda.Universe(PSF, DCD)
u.residues[[]].omega_selections()

```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) develop 2.0.0-dev
- Which version of Python (`python -V`)? 3.7
- Which operating system? MacOS

## Update ##
This has been partially fixed by #3089. The error message for methods such as ``omega_selections()`` is now:

```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF, DCD
>>> u = mda.Universe(PSF, DCD)
>>> u.residues[[]].omega_selections()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mdanalysis/package/MDAnalysis/core/topologyattrs.py", line 1082, in omega_selections
    for r in nxt]
TypeError: 'int' object is not iterable
```

### PR DESCRIPTION
**Pull request by kainszs**
Fixes #2879

Changes made in this Pull Request:
 - the methods `psi_selections()`, `omega_selections()`, `chi1_selections()` can handle empty groups
 - I also noticed a fourth method which fits into the schema and also throws an error with empty groups
    - `phi_selections`

> **_NOTE:_** Since the methods return by default a list of atom groups, the methods will also return an empty list in the case of empty residues instead of an empty residue group ask in the issue. If you accept the code for the change, I will add comments, the tests and update the CHANGELOG.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4529.org.readthedocs.build/en/4529/

### CODE REVIEW

**Comment by yuxuanzhuang** [comment]
Hi @kainszs , welcome to MDAnalysis! Please ping me whenever you need a review for this PR.

**Comment by kainszs** [comment]
Thank you for your offer to review the PR @yuxuanzhuang . What are you still missing for the review? The tests and the changelog? I'll add them tomorrow and get back to you.

**Comment by yuxuanzhuang** [comment]
I noticed the test (main tests) all failed. It indicates something was wrong with your PR (or at least it broke something that worked before). If you click on Details, you will see which tests fail.

**Comment by kainszs** [comment]
Hello @yuxuanzhuang, I think the PR is ready for a review. 

I have noticed that some tests also fail in other PRs. In PR #4524 you mentioned that the linter and main_tests (ubuntu-latest, 3.12, true, true) may fail, but what about codecov/project?

**Comment by yuxuanzhuang** [comment]
The codecov (coverage) also looks fine. Don't worry about the current failed checks as they also seem to be unrelated to your PR. What you also need to add to the PR is

- add yourself to the author list :)
- add an entry in the CHANGELOG (ref on the PR and issue)

**Comment by kainszs** [comment]
Hello @yuxuanzhuang ,

what do I enter under changelog?

Best, Kai

**Comment by yuxuanzhuang** [comment]
@kainszs The comments regarding what to be add in the changelog somehow are not showing up in this PR, but you need to add an entry in package/CHANGELOG. You need to add to the * 2.8.0 entry block; add a reference to the issue and pr (see the others as examples) and your name in line17.

**Comment by kainszs** [comment]
So now I am officially an open source developer. ;)

@yuxuanzhuang thank you for your time.

**Comment by orbeckst** [comment]
@yuxuanzhuang given that you already commented on the PR, can you please review/shepherd the PR? If you have too much to do, please say something/suggest someone else. Thanks!

**Review by yuxuanzhuang — APPROVED** [review]
lgtm! please also fix the conflicts.

File: package/CHANGELOG [hunk 1]
```diff
@@ -16,12 +16,13 @@ The rules for this file:
...
 ??/??/?? IAlibay, HeetVekariya, marinegor, lilyminium, RMeli,
          ljwoods2, aditya292002, pstaerk, PicoCentauri, BFedder,
-         kainszs
+         kainszs, tyler.je.reddy, SampurnaM
 
  * 2.8.0
 
 Fixes
  * Atomname methods can handle empty groups (Issue #2879, PR #4529)
+ * Add support for TPR files produced by Gromacs 2024.1 (PR #4523)
```
Comment (kainszs) [inline package/CHANGELOG:25]: @yuxuanzhuang I noticed a missing reference to the PR, so I added it. I hope that ist okay.

Comment (orbeckst) [reply package/CHANGELOG:25]: Normally you don't have to fix up the CHANGELOG (it confuses reviewers...) but this is fine, thanks to your explanation.

**Comment by orbeckst** [comment]
Failures seem to be due to #4540 / #4209 so I will merge as is.

**Comment by orbeckst** [comment]
Congratulations to your first merged PR @kainszs ! 🎉 

Thank you for your contribution!

**Comment by yuxuanzhuang** [comment]
Congrats! @kainszs

**Pull request merged**